### DISCUSSION TITLE
How can I write request with "or" and "and"?

### DISCUSSION THREAD

**Discussion by danilsizov** [supabase/supabase#6533] [Questions] [answered]
I can't understand how I can filter for example:
"(id.eq.val1 OR id.eq.val2) AND (name.eq.val3 OR id.eq.val4)"

I just checked examples, but I didn't get how it works.

Thank you!

**Comment by GaryAustin1** [comment, marked as answer]
If this is JS client you would just use two .or statements
.or(1,2)
.or(3,4)
there is an implied and

**Comment by saleheen-dev** [comment]
You can filter by using multiple `.eq()`

**For example:**

```
const { data, error } = await supabase
  .from('cities')
  .select('name, country_id')
  .eq("name",'Beijing').eq("country_id", 156}
```


Reference: #4286
